test/dataset/test_rolling.py

```diff
@@ -0,0 +1,306 @@
+"""
+
+from math import floor
+import pandas as pd
+from gluonts.dataset.artificial import constant_dataset
+from gluonts.dataset.common import ListDataset
+from gluonts.dataset.rolling_dataset import (
+    StepStrategy,
+    generate_rolling_datasets,
+)
+import pytest
+import itertools
+
+len_to_truncate = 5
+length_of_roll_window = 5
+
+
+def generate_expected_dataset_unique(prediction_length):
+    """
+    test set for the variation that behaves like:
+
+    [1,2,3,4,5,6,7,8,9] 
+             ^
+        start_time
+    
+    becomes
+    
+    [1,2,3,4,5,6,7,8,9]
+    [1,2,3,4,5,6,7]
+    
+    for 
+    
+    length_of_roll_window = 5
+    prediction_length = 2
+    """
+    # for constant dataset
+    length_timeseries = 30
+    num_timeseries = 10
+    a = []
+    trunc_length = length_timeseries - len_to_truncate
+    for i in range(num_timeseries):
+        iterations = floor(length_of_roll_window / prediction_length)
+        for ii in range(iterations):
+            a.append((float(i), trunc_length - prediction_length * ii))
+
+    return a
+
+
+def generate_expected_dataset_standard(prediction_length):
+    """
+    test set for the variation that behaves like:
+
+    [1,2,3,4,5,6,7,8,9] 
+             ^
+        start_time
+    becomes
+    
+    [1,2,3,4,5,6,7,8,9]
+    [1,2,3,4,5,6,7,8]
+    [1,2,3,4,5,6,7]
+    [1,2,3,4,5,6]
+    
+    for 
+    
+    length_of_roll_window = 5
+    prediction_length = 2
+    """
+    # for constant dataset
+    length_timeseries = 30
+    num_timeseries = 10
+
+    a = []
+    trunc_length = length_timeseries - len_to_truncate
+    for i in range(num_timeseries):
+        for ii in range(length_of_roll_window - prediction_length + 1):
+            a.append((float(i), trunc_length - ii))
+    return a
+
+
+def convert_to_expected(list_of_lists):
+    flattened = list(itertools.chain.from_iterable(list_of_lists))
+    return [(float(0), i) for i in flattened]
+
+
+def generate_expected_dataset_varying_standard(prediction_length):
+    lengths = None
+    if prediction_length == 2:
+        lengths = [
+            [25 - i for i in range(4)],  # test 1
+            [25 - i for i in range(4)],  # test 2
+            [23, 22],  # test 3
+            [],  # test 4
+            [],  # test 5
+            [5 - i for i in range(4)],  # test 6
+            [3, 2],  # test 7
+            [],  # test 8
+            [],  # test 9
+            [3, 2],  # test 10
+        ]
+
+    return convert_to_expected(lengths)
+
+
+def generate_expected_dataset_varying_unique(prediction_length):
+    lengths = None
+    if prediction_length == 2:
+        lengths = [
+            [25, 23],  # test 1
+            [25, 23],  # test 2
+            [23],  # test 3
+            [],  # test 4
+            [],  # test 5
+            [5, 3],  # test 6
+            [3],  # test 7
+            [],  # test 8
+            [],  # test 9
+            [3],  # test 10
+        ]
```

Comment:
I don't like this. Data for test-cases should be defined together, not at different places.